impexp-core/src/main/java/org/citydb/citygml/exporter/database/content/DBSurfaceGeometry.java

```diff
@@ -315,16 +316,16 @@ else if (geomNode.isSolid)
 		}
 
 		// check for xlinks
-		if (geomNode.gmlId != null && geomNode.isXlink) {
+		if (geomNode.gmlId != null && geomNode.isXlink > 0) {
 			if (useXLink) {
-				if (exporter.lookupAndPutGeometryUID(geomNode.gmlId, geomNode.id)) {
+				if (exporter.lookupAndPutGeometryUID(geomNode.gmlId, geomNode.id, geomNode.isXlink == XlinkType.LOCAL.value())) {
 					// check whether we have to embrace the geometry with an orientableSurface
 					return geomNode.isReverse != isSetOrientableSurface ?
 							new SurfaceGeometry(reverseSurface("#" + geomNode.gmlId)) :
 							new SurfaceGeometry("#" + geomNode.gmlId, surfaceGeometryType);
 				}
 			} else {
```

Comment:
There is a small bug caused by the code here. After choosing the option "_perferences -> CityGML export -> Xlinks -> Copy geometry element (use UUID as new gml:id)_", only the gml:id values of the copies should be recreated. However, the gml:id values of the original geometries are also recreated.  
This bug already exists in the current master branch. But i think it could be fixed it in this PR directly.

Reply:
Thanks for spotting this! Reverted 46b5792e200fb613bd57dc863a5f838d53dbfd89 to fix it.